### ISSUE TITLE
Fix error message when writing commandline to memory

### ISSUE THREAD

**Issue by dianpopa** [firecracker-microvm/firecracker#855]
If the statement 
```
guest_mem
        .write_slice_at_addr(cmdline.to_bytes_with_nul(), guest_addr)
        .map_err(|_| Error::CommandLineCopy)?;
```
fails we will be getting an ugly and counterintuitive error message.
**"fault_message": "Cannot load kernel due to invalid memory configuration or invalid kernel image. CommandLineCopy"**
Given that this is more of  a programming logic (trying to write to a memory address that is not mapped) we might as well transform it into an `expect`.

**Comment by lyuyuan92** [comment]
May I take this one please?

**Comment by acatangiu** [comment]
@lyuyuan92 sure, I've assigned it to you.

**Comment by lyuyuan92** [comment]
#971

**Comment by dianpopa** [comment]
After a closer look to this problem additional aspects need to be taken care of.
[All the errors](https://github.com/firecracker-microvm/firecracker/blob/master/kernel/src/loader/mod.rs#L20) that the kernel loader throws will reach the user. That means that whenever the `load_kernel` function and `load_cmdline` fail, the error messages will have the same ugly format.
There are two things to be done here:
1. Implement display for all these errors (similar with [how the logger does it](https://github.com/firecracker-microvm/firecracker/blob/master/logger/src/error.rs#L37))
2. The `load_cmdline` function mistakenly propagates the `Loader` [error](https://github.com/firecracker-microvm/firecracker/blob/master/vmm/src/vmm_config/instance_info.rs#L78) which will throw an inaccurate error message for when we load the cmdline (i.e. "Cannot load *kernel* due to ..."). We need to add another `StartMicrovmError` that displays an accurate message for when we try to load the *command line*.

**Comment by dianpopa** [comment]
Fixed by #971
